app/controllers/user_controller.php

```diff
@@ -73,4 +85,89 @@ public function logout()
+        $user->id = Param::get('user_id');
+        $user->getProfile();
+        $user->isUser($_SESSION['userid']);
+
+        if (!isset($user->username)) {
+            redirect('notfound/pagenotfound');
+        }
+
+        $user->getFollowedThreads($_SESSION['userid']);
+        $user->getCreatedThreads($_SESSION['userid']);
+        $this->set(get_defined_vars());
+    }
+
+    public function edit()
+    {
+        $process = Param::get('process', 'edit');
+        $user = new User();
+
+        switch ($process) {
+            case self::EDIT_ACCOUNT:
+                $user->id = get_authenticated_user_id($_SESSION['userid']);
+                $user->fname = Param::get('firstname');
+                $user->lname = Param::get('lastname');
+                $user->new_username = Param::get('username');
+                $user->new_email    = Param::get('email');
+                try {
+                    $user->updateAccount();
+                    $_SESSION['username'] = $user->new_username;
+                    $user->editSuccess    = true; 
+                } catch(ValidationException $e) {
+                }
+                break;
+            case self::EDIT_PROFILE:
+                $user->id = get_authenticated_user_id($_SESSION['userid']);
+                $user->company = Param::get('company');
+                $user->division = Param::get('division');
+                $user->specialization = Param::get('specialization');
+                try {
+                    $user->updateProfile();
+                    $user->editSuccess = true;
+                } catch(ValidationException $e) {
+                }
+                break;
+            case self::EDIT_PASSWORD:
+                $user->id = get_authenticated_user_id($_SESSION['userid']);
+
+                //set username and old password to password
+                //property to authenticate user
+                $user->username = $_SESSION['username'];
+                $user->password = htmlentities(Param::get('oldPassword'));
+
+                if(!$user->isRegistered()) {
+                    $user->validation_errors['notAuthorized']['authenticate'] = true;
+                    break;
+                }
+                //Unset username so it won't be included in validation
+                unset($user->username);
+                $user->password        = htmlentities(Param::get('password'));
+                $user->confirmpassword = htmlentities(Param::get('confirmPassword'));
+
+                try {
+                    $user->updatePassword();
+                    $user->editSuccess = true;
+                } catch (ValidationException $e) {
+
+                }
+                break;
+            case self::EDIT_PAGE:
+                $user->id = $_SESSION['userid'];
+                break;
+        }
+        
+        $user->getProfile();
+        $this->set(get_defined_vars());
+    }
+
+
+    public function login_end()
+    {
+        
+    }
```

Comment:
Yes po, before I didn't need this function po because I just render the page after login but due to session checking issue I needed to change the url from user/login to user/login_end so it won't be treated as unauthorized access po. :smile: